src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php

```diff
@@ -89,6 +89,7 @@ public function build(ContainerBuilder $container)
 
         $container->addCompilerPass(new LoggerPass(), PassConfig::TYPE_BEFORE_OPTIMIZATION, -32);
         $container->addCompilerPass(new RegisterControllerArgumentLocatorsPass());
+        $container->addCompilerPass(new RegisterControllerArgumentLocatorsPass('argument_resolver.controller_not_registered_as_service'));
```

Comment:
instead of computing the same locator twice, can't we reuse the one computed on the line above?
maybe the pass could take an additional argument to do that? (or any better idea of course)

Reply:
I added a fourth argument in the `RegisterControllerArgumentLocatorsPass`. I'm not sure about what should be tested about this fourth argument. I added one simple test to assert that a reference is passed after running the pass. I don't think this is enough what do you suggest?